fission/src/systems/simulation/wpilib_brain/WPILibBrain.ts

```diff
@@ -112,6 +115,29 @@ export class SimPWM {
     }
 }
 
+export class SimCAN {
+    private constructor() {}
+
+    public static GetDeviceWithID(id: number, type: SimType): any {
+        const id_exp = /.*\[(\d+)\]/g
+        const entries = [...simMap.entries()].filter(([simType, _data]) => simType == type || simType == "SimDevice")
+        for (const [_simType, data] of entries) {
+            for (const key of data.keys()) {
+                let result
+                if ((result = [...key.matchAll(id_exp)]) != undefined) {
+                    if (result.length > 0 && result[0].length > 1) {
+                        const parsed_id = parseInt(result[0][1])
+                        if (parsed_id == id) {
+                            return data.get(key)
+                        }
+                    }
+                }
+            }
```

Comment:
I take issue with how deeply nested this code is.
```suggestion
            for (const key of data.keys()) {
                let result = [...key.matchAll(id_exp)]
                if (result?.length <= 0 || result?[0].length <= 1) continue
                const parsed_id = parseInt(result[0][1])
                if (parsed_id != id) continue
                
                return data.get(key)
            }
```